hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkExportDataSetWriter.java

```diff
@@ -93,12 +98,25 @@ public void readExternal(ObjectInput in) throws IOException, ClassNotFoundExcept
         job.setOutputFormatClass(SparkDataSet.EOutputFormat.class);
         job.getConfiguration().set("mapred.output.dir",directory);
 
-        JavaRDD<V> cached=rdd.cache();
-        long writtenRows=cached.count();
+        SparkContext context = rdd.context();
+        List<Long> recordsWrittenByTasks = Lists.newArrayList();
```

Comment:
I think it would be cleaner to use an AtomicLong and not having to synchronize() on anything.

Reply:
Good point. Thanks @dgomezferro